### PR TITLE
feat(hooks): copy step — glob resolution + file copy to worktree

### LINKED ISSUE #25: vs-05b: Copy step — glob resolution + file copy to worktree
## Parent PRD

#1

## What to build

Implement the hook `copy` step in `src/hooks/`:

- Resolve glob patterns (e.g., `".env*"`) against the main repo checkout using `globset`
- Copy matching files to the worktree root
- Support `!`-prefixed exclusion patterns (e.g., `"!*.example"`)
- Preserve file permissions
- Log each copied file

See FR-21.

### Layers touched
hooks, git, tests

## Acceptance criteria

- [ ] `copy = [".env*"]` copies `.env`, `.env.local` etc. from repo root to worktree
- [ ] `copy = [".env*", "!.env.example"]` excludes `.env.example`
- [ ] Glob patterns resolved via `globset` crate
- [ ] File permissions preserved on copy
- [ ] Each copied file logged (file name + source → destination)
- [ ] No-op if no files match (not an error)
- [ ] Unit tests with temp directories and test files

## Blocked by

- #21

## User stories addressed

- US-2 (auto-copy .env files to new worktrees)

### PR DESCRIPTION
**Pull request by sadiksaifi**
Closes #25

## Summary

Implement the hook `copy` step that resolves glob patterns against the main repo checkout using `globset` and copies matching files to the worktree root. Supports `!`-prefixed exclusion patterns, preserves file permissions, and returns structured `CopyResult` with source→destination info for each copied file. No-op when no files match.

## User Stories Addressed

- US-2: Have my `.env` files and `node_modules` set up automatically in new worktrees

## Automated Testing

- Total tests added: 7
- Tests passing: 7
- Tests failing: 0
- `cargo clippy`: pass (no errors; dead-code warnings are pre-existing and unrelated)
- `cargo test`: pass (265 total tests)

## UAT Checklist

- [ ] Add `copy = [".env*"]` to `[hooks.post_create]` in `.trench.toml`, create `.env` and `.env.local` in repo root, then `trench create test-branch` → both files appear in the new worktree
- [ ] Add `copy = [".env*", "!.env.example"]` to `.trench.toml`, create `.env`, `.env.local`, `.env.example` in repo root, then `trench create test-branch` → `.env.example` is NOT copied
- [ ] Create a shell script with `chmod +x` in repo root, add it to `copy` patterns, create worktree → script retains executable permission in the worktree
- [ ] Configure `copy` patterns that match nothing in the repo → `trench create` succeeds without error (no-op)
- [ ] Add multiple glob patterns (e.g., `copy = ["*.json", "*.toml"]`) → files of both types are copied

## Known Issues

None

## Summary by CodeRabbit

* **New Features**
  * Added file copying functionality with glob pattern matching support for flexible file selection.
  * Supports include and exclude patterns for precise file filtering.
  * Automatically preserves file attributes including executable permissions during copying.

* **Tests**
  * Comprehensive test suite added covering pattern matching, file preservation, and edge cases.

### CODE REVIEW

File: src/hooks/copy.rs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+/// A single file that was copied during the copy step.
+#[derive(Debug, Clone)]
+pub struct CopiedFile {
+    /// File name (relative path from source root).
+    pub name: String,
+    /// Absolute source path.
+    pub source: PathBuf,
+    /// Absolute destination path.
+    pub destination: PathBuf,
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/hooks/copy.rs]: @coderabbitai Fixed in 5baf564 — relaxed the doc comments to reflect that absoluteness depends on the input dirs.

File: src/hooks/copy.rs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            .strip_prefix(root)
+            .context("failed to compute relative path")?;
+
+        if includes.is_match(relative) && !excludes.is_match(relative) {
+            let dest_path = dest_dir.join(relative);
+            if let Some(parent) = dest_path.parent() {
+                std::fs::create_dir_all(parent)?;
+            }
+            std::fs::copy(&path, &dest_path)
+                .with_context(|| format!("failed to copy {} → {}", path.display(), dest_path.display()))?;
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/hooks/copy.rs]: @coderabbitai Fixed in ca97963 — switched to entry.file_type() and explicitly skip symlinks and non-regular files.
